plutus-core/plutus-ir/src/PlutusIR/Contexts.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE LambdaCase #-}
+-- | Datatypes representing 'contexts with holes' in Plutus IR terms.
+--
```

Comment:
Smells like a Zipper!

Reply:
Same idea, yes. I think ultimately we may just want a full one-hole-context type that handles the whole term structure, but that's quite complex.